tests/installation/system_role.pm

```diff
@@ -64,7 +64,7 @@ sub assert_system_role {
     else {
         assert_screen('system-role-default-system', 180);
         my $system_role = get_var('SYSTEM_ROLE', 'default');
-        change_system_role($system_role) if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default'));
+        change_system_role($system_role) if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default') && !check_screen("system-role-$system_role-selected"));
```

Comment:
Hmm this might mean that https://github.com/os-autoinst/os-autoinst/blob/a9af388073889ea5a28f3613d4b40fdfcdee265d/testapi.pm#L1415 might be broken somewhere down the line.

@okurz do you have an idea of what might have caused https://openqa.suse.de/tests/8010789#step/system_role/2?

Reply:
@dzedro Line33 is the culprit of the problem, it should first do `if !check_screen("system-role-$system_role-selected"`

Reply:
Yo umean 37, but not for videomode text ? `check_var('VIDEOMODE', 'text')`

Reply:
Yeah, for the `VIDEOMODE=text` the extra check has to be added too

Reply:
You should be able to just use `match_has_tag` instead of `check_screen` if the needles used in the above `assert_screen` include the tag you look for. Keep in mind that needles can have multiple tags.

Reply:
But they must not e.g. VR https://openqa.suse.de/tests/8013200/modules/system_role/steps/2/edit